src/test/ensemble_config_change_stress.cpp

```diff
@@ -0,0 +1,668 @@
+            ]
+        }
+    ]
+})";
+
+class StressPipelineConfigChanges : public TestWithTempDir {
+    const uint loadThreadCount = 20;
+    const uint beforeConfigChangeLoadTimeMs = 30;
+    const uint afterConfigChangeLoadTimeMs = 50;
+    const int stressIterationsLimit = 50000;
+
+    std::string configFilePath;
+    std::string ovmsConfig;
+    std::string modelPath;
+
+    const std::string pipelineInputName = "custom_dummy_input";
+    const std::string pipelineOutputName = "custom_dummy_output";
+public:
+    void SetUpConfig(const std::string& configContent) {
+        ovmsConfig = configContent;
+        const std::string modelPathToReplace{"/ovms/src/test/dummy"};
+        ovmsConfig.replace(ovmsConfig.find(modelPathToReplace), modelPathToReplace.size(), modelPath);
+        configFilePath = directoryPath + "/ovms_config.json";
+    }
+    void TearDown() override {}
+    void SetUp() override {
+        TestWithTempDir::SetUp();
+        modelPath = directoryPath + "/dummy/";
+        SetUpConfig(stressTestPipelineOneDummyConfig);
+        std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    }
+    void defaultVersionRemove() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        ovmsConfig = stressTestPipelineOneDummyRemovedConfig;
+        createConfigFileWithContent(ovmsConfig, configFilePath);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void defaultVersionAdd() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        std::filesystem::copy("/ovms/src/test/dummy/1", modelPath + "/2", std::filesystem::copy_options::recursive);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void changeToAutoShape() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        ovmsConfig = stressTestPipelineOneDummyConfigChangedToAuto;
+        createConfigFileWithContent(ovmsConfig, configFilePath);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void removePipelineDefinition() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        ovmsConfig = stressTestPipelineOneDummyConfigPipelineRemoved;
+        createConfigFileWithContent(ovmsConfig, configFilePath);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void changeConnectionName() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        ovmsConfig = stressTestPipelineOneDummyConfigChangeConnectionName;
+        createConfigFileWithContent(ovmsConfig, configFilePath);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void addNewPipeline() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        ovmsConfig = stressTestPipelineOneDummyConfigAddNewPipeline;
+        createConfigFileWithContent(ovmsConfig, configFilePath);
+        SPDLOG_INFO("{} end", __FUNCTION__);
+    }
+    void retireSpecificVersionUsed() {
+        SPDLOG_INFO("{} start", __FUNCTION__);
+        std::filesystem::copy("/ovms/src/test/dummy/1", modelPath + "/2", std::filesystem::copy_options::recursive);
```

Comment:
Yes because I'am using latest model version policy, latest=1. I could either add new version or have two versions loaded and remove 1 of them. Both ways achieves the same result -> unloadModel is called on the version used in pipeline.